pennylane_qiskit/qiskit_device2.py

```diff
@@ -239,10 +239,18 @@ class QiskitDevice2(Device):
         "PauliX",
         "PauliY",
         "PauliZ",
-        "Identity",
         "Hadamard",
         "Hermitian",
+        "Identity",
         "Projector",
+        "SparseHamiltonian",
+        "Hamiltonian",
+        "LinearCombination",
+        "Sum",
+        "SProd",
+        "Prod",
+        "Exp",  # this may not work
+        "Evolution",  # this may not work
```

Comment:
What is the motivation for putting these here? `Exp` isn't really an observable, and I'm a little hesitant to add `Evolution` to the list of supported observables, especially without tests

Reply:
Ah, no real motivation here, I just copy and pasted the supported observables from the default qubit and wrote these two notes to myself and then forgot about it 😓 . You're right though, they're not observables they're operators. And as operators they are not supported either so I will delete this.

Reply:
I also removed SparseHamiltonian for now, as it is only supported for Expval and not Var and Samples.

Reply:
They should both, when they are based on appropriate inputs, be decomposed to something that the device supports - they are listed on default.qubit because as a simulator, it can support them natively.

I'm actually a little hesitant to depart from the observables included in the current QiskitDevice, i.e. 

```
    observables = {
        "PauliX",
        "PauliY",
        "PauliZ",
        "Identity",
        "Hadamard",
        "Hermitian",
        "Projector",
    }
```
at least without testing them using both Estimator, Sampler and the old `run` function on a hardware device. 

Things like `Hamiltonian` currently work on the old version of the Qiskit device because they are broken down by `hamiltonian_expand`. It should be compatible with Estimator specifically, but what about Sampler? 

Also out of curiosity, does your test involving `Prod` fail if you remove `Prod` from this list?

Reply:
> Also out of curiosity, does your test involving Prod fail if you remove Prod from this list?

to clarify, not the stopping condition test, but the integration tests

Reply:
> to clarify, not the stopping condition test, but the integration tests

So removing the stopping condition and removing Prod from the list, the integration tests pass without issue.

Reply:
> They should both, when they are based on appropriate inputs, be decomposed to something that the device supports - they are listed on default.qubit because as a simulator, it can support them natively.
> 
> I'm actually a little hesitant to depart from the observables included in the current QiskitDevice, i.e.
> 
> ```
>     observables = {
>         "PauliX",
>         "PauliY",
>         "PauliZ",
>         "Identity",
>         "Hadamard",
>         "Hermitian",
>         "Projector",
>     }
> ```
> 
> at least without testing them using both Estimator, Sampler and the old `run` function on a hardware device.
> 
> Things like `Hamiltonian` currently work on the old version of the Qiskit device because they are broken down by `hamiltonian_expand`. I think it should be compatible with Estimator specifically without breaking it down like that, but what about Sampler?

In this case, we could restrict things like 'Hamiltonian' to only work with Expval, the same way 'SparseHamiltonian' is restricted on the defaultqubit device. Alternatively I can add tests for the Sampler and the old run function as well, although that to me is out of scope for this generalization of mp_to_pauli. That to me would be a separate PR i.e. "Ensure/make all observables compatible with all measurement processes". What do you think?

Reply:
> They should both, when they are based on appropriate inputs, be decomposed to something that the device supports - they are listed on default.qubit because as a simulator, it can support them natively.
> 

Can you clarify further what you mean by this? Do you mean to say that there _is_ a pauli_rep for these observables that I can use to convert into an SPO? Or just that theoretically there should exist a pauli_rep that is just not implemented right now?

Reply:
Don't worry about the `pauli_rep` in the context of the set of allowed `observables`. The device can support lots of observables on the user end that aren't listed here. This is a list of what is supported on the device at the time the device runs `execute`. Most things have `decomposition` methods defined, and can be decomposed into things that the device supports at execution, but some things can't. Though now that I'm giving it more thought, I'm not sure we can decompose `Exp` specifically to be a valid observable for this device. But we can run many things that aren't on the `observables` list.

Wrt. how to deal with Hamiltonian, it shouldn't be necessary to restrict what type of measurement it works with once the transform is added to preprocessing. That is planned to happen in [this story](https://app.shortcut.com/xanaduai/story/54392/transforms-are-included-in-preprocessing-for-observables-that-require-hamiltonian-expand-sum-expand-and-grouping, doesn't involve adding `Hamiltonian` to supported observables.

This might be easier to discuss in person, should we try to chat sometime tomorrow?

Reply:
Yes, let's discuss in person instead. That would probably be easier 😅 .